pkg/edgedictionaryitem/update.go

```diff
@@ -0,0 +1,54 @@
+package edgedictionaryitem
+
+import (
+	"io"
+
+	"github.com/fastly/cli/pkg/common"
+	"github.com/fastly/cli/pkg/compute/manifest"
+	"github.com/fastly/cli/pkg/config"
+	"github.com/fastly/cli/pkg/errors"
+	"github.com/fastly/cli/pkg/text"
+	"github.com/fastly/go-fastly/v2/fastly"
+)
+
+// UpdateCommand calls the Fastly API to describe a service.
+type UpdateCommand struct {
+	common.Base
+	manifest manifest.Data
+	Input    fastly.UpdateDictionaryItemInput
+
+	itemvalue common.OptionalString
+}
+
+// NewUpdateCommand returns a usable command registered under the parent.
+func NewUpdateCommand(parent common.Registerer, globals *config.Data) *UpdateCommand {
+	var c UpdateCommand
+	c.Globals = globals
+	c.manifest.File.Read(manifest.Filename)
+	c.CmdClause = parent.Command("update", "Update or insert an item on a Fastly edge dictionary")
+	c.CmdClause.Flag("service-id", "Service ID").Short('s').StringVar(&c.manifest.Flag.ServiceID)
+	c.CmdClause.Flag("dictionary-id", "Dictionary ID").Required().StringVar(&c.Input.DictionaryID)
+	c.CmdClause.Flag("key", "Dictionary item key").Required().StringVar(&c.Input.ItemKey)
+	c.CmdClause.Flag("value", "Dictionary item value").Required().Action(c.itemvalue.Set).StringVar(&c.itemvalue.Value)
```

Comment:
So annoyingly the Go-Fastly implementation has the item 'value' as an optional field although I believe that to be a mistake and in fact it should be a required field (because why bother calling the api without an updated value for the key -- I don't believe it's even possible to set an empty string as a value for an edge dict key either).

For now let's keep the `.Required()` as we want the command to be useful (i.e. don't allow someone to make request without a value) but we'll still need the `common.OptionalString` for now as the version of Go-Fastly we have is internally using a pointer. When I eventually cut a new Go-Fastly release I'll come back (as per the `TODO(integralist)` comment above) and fix up this section.